### PR TITLE
HEE-124: Newsletter Subscription

### PR DESCRIPTION
**Pull request by thojava**
This PR is going to setup the foundation to build the form for website in general and create Newsletter Subscription form specifically. 
We integrated Enterprise Forms plugin https://documentation.bloomreach.com/14/library/enterprise/enterprise-features/enterprise-forms/enterprise-forms.html which provide the editor the abilities to control the form elements and submission behavior directly in the CMS.
Form page prototypes also being created to allow editor create an simple form page directly from Experience Manager.

Newsletter Subscription form was created and can be accessed from local URL http://localhost:8080/site/newsletter-subscription 

The form thank you page will be created on other PR.

### CODE REVIEW

File: site/webapp/src/main/webapp/static/js/jquery-hippo-validate.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2014 Hippo B.V. (http://www.onehippo.com)
+ */
+
+jQuery.validateExtend({
```
Comment (duncanheron) [inline site/webapp/src/main/webapp/static/js/jquery-hippo-validate.js:5]: @thojava does Bloomreach suggest using jQuery?

Comment (thojava) [reply site/webapp/src/main/webapp/static/js/jquery-hippo-validate.js:5]: @duncanheron I don't think Bloomreach have any preference on this. But the Enterprise Form plugin sample code using Jquery so I followed that.

Comment (thojava) [reply site/webapp/src/main/webapp/static/js/jquery-hippo-validate.js:5]: @duncanheron I don't think Bloomreach have any preference on this. But the Enterprise Form plugin sample code using Jquery so I followed that.

Comment (duncanheron) [reply site/webapp/src/main/webapp/static/js/jquery-hippo-validate.js:5]: Ok if the form plugin uses that convention thats fine - leave this as is. 👍

**Comment by smanivasag** [comment]
Thanks @thojava. Do you think it's worth enabling validation and autocomplete for the fields ?

Guess the submission of form data to `pure360` will be done part of subscription thank you page ticket ?

Just out of interest, are `repository-data/webfiles/src/main/resources/site/freemarker/hee/eforms/eforms-field-renderer.ftl` & `repository-data/webfiles/src/main/resources/site/freemarker/hee/eforms/eforms-validation.ftl` being used internally by Enterprise Forms plugin ? Sorry, doesn't know much about Enterprise Forms plugin. Thank you!

**Comment by thojava** [comment]
@smanivasag In this plugin, we have the `validation` and `autocomplete` feature which editor can setup directly in the CMS.
In the sample form, I have just enable the autocomplete on these 2 fields and mark email validation on the Email field.

About `eforms-field-renderer.ftl` and `eforms-validation.ftl`, they are being used by the sample code which provided by Bloomreach, not officially but something we can use and customize on by our own.

**Review by smanivasag — APPROVED** [review]
> @smanivasag In this plugin, we have the `validation` and `autocomplete` feature which editor can setup directly in the CMS.
> In the sample form, I have just enable the autocomplete on these 2 fields and mark email validation on the Email field.
> 
> About `eforms-field-renderer.ftl` and `eforms-validation.ftl`, they are being used by the sample code which provided by Bloomreach, not officially but something we can use and customize on by our own.

Sure, thanks for the details @thojava

File: repository-data/site/src/main/resources/hcm-content/hst/configurations/lks/workspace/sitemap/newsletter-subscription.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/hst:hst/hst:configurations/lks/hst:workspace/hst:sitemap/newsletter-subscription:
```
[the remark this answers is not in the document]
Comment (thojava) [reply repository-data/site/src/main/resources/hcm-content/hst/configurations/lks/workspace/sitemap/newsletter-subscription.yaml]: @smanivasag Yeap I just do it, by default system export to the site module. We should be able to find a way to configure the export location. I am reading on this https://documentation.bloomreach.com/14/library/development/automatic-export-add-on.html part `Configuring AutoExport Modules` but not be able to make it ;)

[the remark this answers is not in the document]
Comment (smanivasag) [reply repository-data/site/src/main/resources/hcm-content/hst/configurations/lks/workspace/sitemap/newsletter-subscription.yaml]: @thojava - Thanks for checking on this, looks like this is bit tricky. Worth raising this in Bloomreach Community ?

**Pull request merged**